src/cmd/src/datanode.rs

```diff
@@ -139,9 +139,11 @@ impl TryFrom<StartCommand> for DatanodeOptions {
             opts.wal.dir = wal_dir;
         }
 
-        if let Some(procedure_dir) = cmd.procedure_dir {
-            opts.procedure = Some(ProcedureConfig::from_file_path(procedure_dir));
-        }
+        opts.procedure = if let Some(path) = cmd.procedure_dir {
```

Comment:
issue: This changes the meaning of `procedure_dir`, which means the places to store procedure data. It also enables the procedure framework by default.

I'd suggest reverting all changes introduced to this file or removing the `procedure_dir` argument. It is recommended to configure the procedure framework via the config file.